src/main/java/com/bittercode/service/impl/UserServiceImpl.java

```diff
@@ -14,8 +14,12 @@
 import com.bittercode.model.UserRole;
 import com.bittercode.service.UserService;
 import com.bittercode.util.DBUtil;
+import org.checkerframework.checker.nullness.qual.Nullable;
 
-public class UserServiceImpl implements UserService {
+
+// Suppressing warnings because the only time "register" method is used is when all used fields of
+// The user parameter are NonNull due to setter method implementations added.
```

Comment:
In general, you should try to limit the scope of warning suppressions. It's almost never necessary to scope them to a whole class: instead, it's better to suppress a warning at a particular method (or even better, line).

Also, you should fully-specify the warning key to suppress. In this case, you've suppressed all `contracts.precondition` warnings, which would include those from other Checker Framework checkers if they were ever to be run on this code. Instead, you should suppress `nullness:contracts.precondition` only.